internal/client/client.go

```diff
@@ -850,3 +850,48 @@ func SplitMxc(mxcUri string) (string, string) {
 
 	return origin, mediaId
 }
+
+// SendToDeviceMessages sends to-device messages over /sendToDevice/.
+//
+// The messages parameter is nested as follows:
+// user_id -> device_id -> content (map[string]interface{})
+func (c *CSAPI) SendToDeviceMessages(t *testing.T, evType string, messages map[string]map[string]map[string]interface{}) {
```

Comment:
Given this is a helper function, it would be nice to have `SendToDeviceMessage` (singular not plural) which has a nicer API shape:
```go
func (c *CSAPI) SendToDeviceMessage(t *testing.T, evType, userID, deviceID string, content interface{}) {
```

This can then call through to this function. For most people, they just want to send a to-device message to one device or `*` devices for a single user, which the simpler format makes clearer.

Reply:
I originally had this, but I over-engineered it and added 2 additional "nice" functions for just one test, so I thought that was overkill.